auth.go

```diff
@@ -1,102 +1,111 @@
-	RegisterCircuit(AuthCircuitID, &AuthCircuit{})
+// authCircuitInputs type reflect auth.circom private inputs required by prover
+type authCircuitInputs struct {
+	UserAuthClaim               *core.Claim      `json:"userAuthClaim"`
+	UserAuthClaimMtp            []string         `json:"userAuthClaimMtp"`
+	UserAuthClaimNonRevMtp      []string         `json:"userAuthClaimNonRevMtp"`
+	UserAuthClaimNonRevMtpAuxHi *merkletree.Hash `json:"userAuthClaimNonRevMtpAuxHi"`
+	UserAuthClaimNonRevMtpAuxHv *merkletree.Hash `json:"userAuthClaimNonRevMtpAuxHv"`
+	UserAuthClaimNonRevMtpNoAux string           `json:"userAuthClaimNonRevMtpNoAux"`
+	Challenge                   string           `json:"challenge"`
+	ChallengeSignatureR8X       string           `json:"challengeSignatureR8x"`
+	ChallengeSignatureR8Y       string           `json:"challengeSignatureR8y"`
+	ChallengeSignatureS         string           `json:"challengeSignatureS"`
+	UserClaimsTreeRoot          *merkletree.Hash `json:"userClaimsTreeRoot"`
+	UserID                      string           `json:"userID"`
+	UserRevTreeRoot             *merkletree.Hash `json:"userRevTreeRoot"`
+	UserRootsTreeRoot           *merkletree.Hash `json:"userRootsTreeRoot"`
+	UserState                   *merkletree.Hash `json:"userState"`
 }
 
-// GetVerificationKey returns key to verify proof
-func (c *AuthCircuit) GetVerificationKey() VerificationKeyJSON {
-	return AuthenticationVerificationKey
-}
+// CircuitInputMarshal returns Circom private inputs for auth.circom
+func (a AuthInputs) CircuitInputMarshal() ([]byte, error) {
+
+	s := authCircuitInputs{
+		UserAuthClaim: a.AuthClaim.Claim,
+		UserAuthClaimMtp: PrepareSiblingsStr(a.AuthClaim.Proof.AllSiblings(),
+			a.GetMTLevel()),
+		UserAuthClaimNonRevMtp: PrepareSiblingsStr(a.AuthClaim.NonRevProof.Proof.AllSiblings(),
+			a.GetMTLevel()),
+		Challenge:             a.Challenge.String(),
+		ChallengeSignatureR8X: a.Signature.R8.X.String(),
+		ChallengeSignatureR8Y: a.Signature.R8.Y.String(),
+		ChallengeSignatureS:   a.Signature.S.String(),
+		UserClaimsTreeRoot:    a.AuthClaim.TreeState.ClaimsRoot,
+		UserID:                a.ID.BigInt().String(),
+		UserRevTreeRoot:       a.AuthClaim.TreeState.RevocationRoot,
+		UserRootsTreeRoot:     a.AuthClaim.TreeState.RootOfRoots,
+		UserState:             a.AuthClaim.TreeState.State,
+	}
+
+	nodeAuxAuth := getNodeAuxValue(a.AuthClaim.Proof.NodeAux)
+	s.UserAuthClaimNonRevMtpAuxHi = nodeAuxAuth.key
+	s.UserAuthClaimNonRevMtpAuxHv = nodeAuxAuth.value
+	s.UserAuthClaimNonRevMtpNoAux = nodeAuxAuth.noAux
 
-// GetPublicSignalsSchema returns schema to parse public inputs
-func (c AuthCircuit) GetPublicSignalsSchema() PublicSchemaJSON {
-	return AuthenticationPublicSignalsSchema
+	return json.Marshal(s)
 }
 
-// PrepareInputs returns inputs for id state circuit as map
-func (c *AuthCircuit) PrepareInputs(in TypedInputs) (map[string]interface{}, error) {
+// AuthPubSignals auth.circom public signals
+type AuthPubSignals struct {
```

Comment:
It looks confusing how we use "signal" and "input" terms. "Signal" means everything which can be

- private input
- public input
- output
I thing we should align the naming with it